.github/workflows/regression-tests.yml

```
@@ -2,11 +2,45 @@ name: Regression Tests
 
 on:
   workflow_dispatch:
+    inputs:
+      branchName:
+        description: 'Branch Name'
+        required: true
+        default: 'main'
+  pull_request_target:
+    types: [labeled]
 
 jobs:
-  test:
+  run-regression-test-on-sqlite:
+    if: (github.event_name == 'workflow_dispatch') || (github.event_name == 'pull_request_target' && github.event.label.name == 'regression-tests')
     runs-on: ubuntu-latest
     steps:
-      - name: Echo Hello World
-        run: |
-          echo "Hello World"
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branchName || github.head_ref }}
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '^1.21.5'
```

use `go-version-file` instead:
```suggestion
        with:
          go-version-file: 'go.mod'
```

---

done